src/uct/ib/mlx5/rc/rc_mlx5_iface.c

```diff
@@ -817,7 +817,8 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_common_t, uct_iface_ops_t *tl_ops,
 
     if ((rc_config->fence_mode == UCT_RC_FENCE_MODE_WEAK) ||
         ((rc_config->fence_mode == UCT_RC_FENCE_MODE_AUTO) &&
-         (uct_ib_device_has_pci_atomics(dev) || md->super.relaxed_order))) {
+         (uct_ib_device_has_pci_atomics(dev) || md->super.relaxed_order ||
+          (self->config.put_fence_flag != 0)))) {
```

Comment:
1. can we put config.put_fence_flag and config.atomic_fence_flag initialization in one place because now it depends on init ordering which is easy to break
2. should atomic fence flag be UCT_IB_MLX5_WQE_CTRL_FLAG_STRONG_ORDER in some cases?